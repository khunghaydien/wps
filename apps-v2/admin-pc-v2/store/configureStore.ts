import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer from '../reducers';

const PRODUCTION = process.env.NODE_ENV === 'production';

const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
        traceLimit: 25,
      })
    : compose;

export default function configureStore(initialState: any) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION) {
    middlewares.push(logger);
  }

  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../reducers', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../reducers').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
