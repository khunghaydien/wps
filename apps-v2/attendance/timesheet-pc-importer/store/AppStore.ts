import { Action, Store } from 'redux';
import { ThunkDispatch } from 'redux-thunk';

import { State } from '@attendance/timesheet-pc-importer/modules';

export type AppState = State;
export type AppStore = Store<State>;
export type AppDispatch = ThunkDispatch<unknown, void, Action<string>>;
