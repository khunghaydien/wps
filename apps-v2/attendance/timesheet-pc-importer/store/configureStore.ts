/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
// 本来は使いたくないが
// Global Container の社員切替が使用しているので
// しかたなく導入している
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.TEST === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middleWares: Middleware[] = [
    // 本来は使いたくないが
    // Global Container の社員切替が使用しているので
    // しかたなく導入している
    thunkMiddleware,
  ];
  if (!PRODUCTION && !TEST) {
    middleWares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middleWares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
