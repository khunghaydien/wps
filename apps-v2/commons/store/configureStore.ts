import { AnyAction, applyMiddleware, createStore, Store } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../reducers';

const createStoreWithMiddleware = applyMiddleware(
  thunkMiddleware,
  logger
)(createStore);

export default function configureStore(
  initialState?: State
): Store<State, AnyAction> {
  const store = createStoreWithMiddleware(rootReducer, initialState);

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../reducers', () => {
      /* eslint-disable  @typescript-eslint/no-var-requires */
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const nextRootReducer = require('../reducers');

      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
}
