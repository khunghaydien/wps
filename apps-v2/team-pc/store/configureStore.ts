/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(initialState?: State) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION) {
    middlewares.push(logger);
  }

  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
