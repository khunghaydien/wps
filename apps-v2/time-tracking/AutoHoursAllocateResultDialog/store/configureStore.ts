/* eslint-disable global-require, import/newline-after-import */
import {
  AnyAction,
  applyMiddleware,
  compose,
  createStore,
  Middleware,
  Store,
} from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer, { State } from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.NODE_ENV === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

export default function configureStore(
  initialState?: Partial<State>
): Store<State, AnyAction> {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION && !TEST) {
    middlewares.push(logger);
  }
  const store: Store<State, AnyAction> = createStore(
    rootReducer,
    initialState,
    // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      /* eslint-disable-next-line @typescript-eslint/no-var-requires */
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
