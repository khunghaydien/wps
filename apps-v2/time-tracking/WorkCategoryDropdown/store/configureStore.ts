/* eslint-disable global-require, import/newline-after-import */
import { applyMiddleware, compose, createStore, Middleware } from 'redux';
import logger from 'redux-logger';
import thunkMiddleware from 'redux-thunk';

import rootReducer from '../modules';

const PRODUCTION = process.env.NODE_ENV === 'production';
const TEST = process.env.NODE_ENV === 'test';

/* eslint-disable no-underscore-dangle */
const composeEnhancers =
  !PRODUCTION && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        name: 'WorkCategoryDropdown',
      })
    : compose;

export default function configureStore(initialState?: any) {
  const middlewares: Middleware[] = [thunkMiddleware];
  if (!PRODUCTION && !TEST) {
    middlewares.push(logger);
  }
  const store = createStore(
    rootReducer,
    initialState, // Redux DevTools Extension
    composeEnhancers(applyMiddleware(...middlewares))
  );

  if (!PRODUCTION && module.hot) {
    module.hot.accept('../modules', () => {
      // eslint-disable-next-line @typescript-eslint/no-var-requires
      const newRootReducer = require('../modules').default;
      store.replaceReducer(newRootReducer);
    });
  }

  return store;
}
