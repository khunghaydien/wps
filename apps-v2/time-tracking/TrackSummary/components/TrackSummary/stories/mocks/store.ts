import defaultPermission from '@apps/domain/models/access-control/Permission';

import configureStore from '../../../../store/configureStore';

const state = {
  common: {
    accessControl: {
      permission: {
        ...defaultPermission,
      },
    },
  },
  entities: {
    summary: {
      targetDate: '2019-7-20',
      content: {
        useRequest: true,
        targetDate: '2019-7-20',
        startDate: '2019-7-16',
        endDate: '2019-8-15',
        request: {
          status: 'NotRequested',
          approverName: 'TeamSpirit',
        },
        taskSummaryRecords: [
          {
            jobName: 'R06(19/07)_WSP_Entire',
            workCategoryName: 'Design',
            workTimeRatio: 62.0,
            workTime: 5829,
            isEditLocked: false,
          },
          {
            jobName: 'R06(19/07)_WSP_Entire',
            workCategoryName: 'Meeting&Others',
            workTimeRatio: 23.0,
            workTime: 2160,
            isEditLocked: false,
          },
          {
            jobName: '（PD-Common）Others',
            workCategoryName: undefined,
            workTimeRatio: 12.3,
            workTime: 990,
            isEditLocked: false,
          },
          {
            jobName: '（PD-Common）Recruting',
            workCategoryName: undefined,
            workTimeRatio: 2.1,
            workTime: 210,
            isEditLocked: false,
          },
          {
            jobName: '（PD-Common）Meeting',
            workCategoryName: 'ミーティング他',
            workTimeRatio: 0.1,
            workTime: 30,
            isEditLocked: false,
          },
        ],
      },
    },
  },
  userSetting: {
    photoUrl:
      'https://www.guidedogs.org/wp-content/uploads/2015/05/Dog-Im-Not.jpg',
  },
};

// @ts-ignore
export const store = configureStore({ initialState: state });

export const emptyDataStore = configureStore({
  initialState: {
    ...state,
    entities: {
      ...state.entities,
      summary: {
        targetDate: '2019-7-20',
        // @ts-ignore
        content: {
          ...state.entities.summary.content,
          taskSummaryRecords: [],
        },
      },
    },
  },
});

export const manyDataStore = configureStore({
  initialState: {
    ...state,
    entities: {
      ...state.entities,
      summary: {
        targetDate: state.entities.summary.targetDate,
        content: {
          ...state.entities.summary.content,
          // @ts-ignore
          taskSummaryRecords: [
            ...state.entities.summary.content.taskSummaryRecords,
            ...state.entities.summary.content.taskSummaryRecords,
            ...state.entities.summary.content.taskSummaryRecords,
            ...state.entities.summary.content.taskSummaryRecords,
          ],
        },
      },
    },
  },
});

export const longDataStore = configureStore({
  initialState: {
    ...state,
    entities: {
      ...state.entities,
      summary: {
        targetDate: '2019-7-20',
        content: {
          ...state.entities.summary.content,
          taskSummaryRecords: [
            // @ts-ignore
            {
              jobName:
                'But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally encounter consequences that are extremely painful. Nor again is there anyone who loves or pursues or desires to obtain pain of itself, because it is pain, but because occasionally circumstances occur in which toil and pain can procure him some great pleasure. To take a trivial example, which of us ever undertakes laborious physical exercise, except to obtain some advantage from it? But who has any right to find fault with a man who chooses to enjoy a pleasure that has no annoying consequences, or one who avoids a pain that produces no resultant pleasure? On the other hand, we denounce with righteous indignation and dislike men who are so beguiled and demoralized by the charms of pleasure of the moment, so blinded by desire, that they cannot foresee',
              workCategoryName: 'Design',
              workTimeRatio: 50.0,
              workTime: 5829,
              isEditLocked: false,
            },
            // @ts-ignore
            {
              jobName: 'R06(19/07)_WSP_Entire',
              workCategoryName: 'Meeting&Others',
              workTimeRatio: 50.0,
              workTime: 2160,
              isEditLocked: false,
            },
          ],
        },
      },
    },
  },
});

export const requestedDataStore = configureStore({
  initialState: {
    ...state,
    entities: {
      ...state.entities,
      summary: {
        targetDate: state.entities.summary.targetDate,
        content: {
          ...state.entities.summary.content,
          // @ts-ignore
          request: {
            ...state.entities.summary.content.request,
            status: 'Pending',
          },
        },
      },
    },
  },
});
