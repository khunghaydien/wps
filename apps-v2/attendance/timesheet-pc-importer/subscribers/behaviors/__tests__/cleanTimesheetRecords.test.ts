import configureMockStore from 'redux-mock-store';

import behavior from '../cleanTimesheetRecords';
import { AppStore } from '@attendance/timesheet-pc-importer/store/AppStore';

const createStore = configureMockStore();

beforeEach(() => {
  jest.clearAllMocks();
});

it('should do', async () => {
  // Arrange
  const store = createStore({
    timesheet: {
      startDate: 'startDate',
      endDate: 'endDate',
    },
  });

  // Act
  await behavior(store as AppStore)();

  // Assert
  expect(store.getActions()).toMatchSnapshot();
});
