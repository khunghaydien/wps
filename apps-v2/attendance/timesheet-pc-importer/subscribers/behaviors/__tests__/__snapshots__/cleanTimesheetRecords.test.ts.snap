// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should do 1`] = `
Array [
  Object {
    "payload": Object {
      "endDate": "endDate",
      "startDate": "startDate",
    },
    "type": "TIMESHEET-PC-IMPORTER/DAILY_RECORDS/CREATE",
  },
]
`;
